generator68k: fix parallel build

rofl0r opened the pull request:
fixes dependency chain for generated headers:

```
Making all in generator68k
make[3]: Entering directory `/home/rofl/tmp/gngeo/src/generator68k'
gcc -c def68k.c -o def68k.o -I .. -I ../.. -I . `sdl2-config --cflags` -Umain -Dmain=main
gcc -c tab68k2.c -o tab68k2.o -I .. -I ../.. -I . `sdl2-config --cflags` -Umain -Dmain=main
gcc -c gen68k.c -o gen68k.o -I .. -I ../.. -I . `sdl2-config --cflags` -Umain -Dmain=main
gen68k.c:8:25: fatal error: def68k-iibs.h: No such file or directory
 #include "def68k-iibs.h"
                         ^
```

also use a stamp file for files generated by gen68k so it isn't invoked
once per generated source file, with each process overwriting each others
output.

remake -x output:

```
          Prerequisite `gen68k' is newer than target `cpu68k-0.c'.
          /tmp/gngeo/src/generator68k/Makefile:663      Must remake target `cpu68k-0.c'.
Invoking recipe from Makefile:664 to update target `cpu68k-0.c'.
 ##>>>>>>>>>>>>>>>>>>>>>>>>>>0>>>>>>>>>>>>>>>>>>>>>>>>>>>>
./gen68k 2>&1 >/dev/null
 ##<<<<<<<<<<<<<<<<<<<<<<<<<<0<<<<<<<<<<<<<<<<<<<<<<<<<<<<
        /tmp/gngeo/src/generator68k/Makefile:666        File `cpu68k-1.o' does not exist.
          Prerequisite `gen68k' is newer than target `cpu68k-1.c'.
          /tmp/gngeo/src/generator68k/Makefile:663      Must remake target `cpu68k-1.c'.
Invoking recipe from Makefile:664 to update target `cpu68k-1.c'.
 ##>>>>>>>>>>>>>>>>>>>>>>>>>>0>>>>>>>>>>>>>>>>>>>>>>>>>>>>
./gen68k 2>&1 >/dev/null
```

rofl0r committed:
generator68k: fix parallel build

fixes dependency chain for generated headers:

Making all in generator68k
make[3]: Entering directory `/home/rofl/tmp/gngeo/src/generator68k'
gcc -c def68k.c -o def68k.o -I .. -I ../.. -I . `sdl2-config --cflags` -Umain -Dmain=main
gcc -c tab68k2.c -o tab68k2.o -I .. -I ../.. -I . `sdl2-config --cflags` -Umain -Dmain=main
gcc -c gen68k.c -o gen68k.o -I .. -I ../.. -I . `sdl2-config --cflags` -Umain -Dmain=main
gen68k.c:8:25: fatal error: def68k-iibs.h: No such file or directory
 #include "def68k-iibs.h"
                         ^
also use a stamp file for files generated by gen68k so it isn't invoked
once per generated source file, with each process overwriting each others
output.

remake -x output:
          Prerequisite `gen68k' is newer than target `cpu68k-0.c'.
          /tmp/gngeo/src/generator68k/Makefile:663      Must remake target `cpu68k-0.c'.
Invoking recipe from Makefile:664 to update target `cpu68k-0.c'.
 ##>>>>>>>>>>>>>>>>>>>>>>>>>>0>>>>>>>>>>>>>>>>>>>>>>>>>>>>
./gen68k 2>&1 >/dev/null
 ##<<<<<<<<<<<<<<<<<<<<<<<<<<0<<<<<<<<<<<<<<<<<<<<<<<<<<<<
        /tmp/gngeo/src/generator68k/Makefile:666        File `cpu68k-1.o' does not exist.
          Prerequisite `gen68k' is newer than target `cpu68k-1.c'.
          /tmp/gngeo/src/generator68k/Makefile:663      Must remake target `cpu68k-1.c'.
Invoking recipe from Makefile:664 to update target `cpu68k-1.c'.
 ##>>>>>>>>>>>>>>>>>>>>>>>>>>0>>>>>>>>>>>>>>>>>>>>>>>>>>>>
./gen68k 2>&1 >/dev/null

dciabrin commented on src/generator68k/Makefile.am:
```diff
@@ -1,43 +1,29 @@
+def68k_SOURCES = def68k.c tab68k2.c
+
+def68k.stamp: def68k$(EXEEXT) def68k.def
 	$(AM_V_GEN)./def68k$(EXEEXT) 2>&1 >/dev/null
+	touch def68k.stamp
```
Can you make it a single line with e.g. "<cmd1> && <cmd2>" to not run touch if def68k failed?

dciabrin commented on src/generator68k/Makefile.am:
```diff
@@ -93,74 +67,13 @@ CPU_OBJ    = cpu68k-0.o  cpu68k-1.o  cpu68k-2.o  cpu68k-3.o  \
 
 $(CPU_OBJ) : %.o : %.c
 	$(AM_V_CC)$(COMPILE) -c $< -o $@
 
```
Could you make sure that the stamp files get deleted with make clean with something like:
```
CLEANFILES = gen68k.stamp def68k.stamp
```

Also those two files should be in .gitignore

rofl0r committed:
parallel build: address PR feedback

- add .stamp files to .gitignore
- add .stamp files to make clean target
- create stamp files only when parent command succeeds

rofl0r commented:
done.

dciabrin commented:
Thanks @rofl0r
